sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -181,3 +187,239 @@ case class Randn(child: Expression, hideSeed: Boolean = false) extends RDG {
 object Randn {
   def apply(seed: Long): Randn = Randn(Literal(seed, LongType))
 }
+
+@ExpressionDescription(
+  usage = """
+    _FUNC_(min, max[, seed]) - Returns a random value with independent and identically
+      distributed (i.i.d.) values with the specified range of numbers. The random seed is optional.
+      The provided numbers specifying the minimum and maximum values of the range must be constant.
+      If both of these numbers are integers, then the result will also be an integer. Otherwise if
+      one or both of these are floating-point numbers, then the result will also be a floating-point
+      number.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(10, 20) > 0 AS result;
+      true
+  """,
+  since = "4.0.0",
+  group = "math_funcs")
+object UniformExpressionBuilder extends ExpressionBuilder {
+  override def build(funcName: String, expressions: Seq[Expression]): Expression = {
+    val numArgs = expressions.length
+    expressions match {
+      case Seq(min, max) =>
+        Uniform(min, max)
+      case Seq(min, max, seed) =>
+        Uniform(min, max, seed)
+      case _ =>
+        throw QueryCompilationErrors.wrongNumArgsError(funcName, Seq(2, 3), numArgs)
+    }
+  }
+}
+
+case class Uniform(min: Expression, max: Expression)
+  extends RuntimeReplaceable with BinaryLike[Expression] with ExpressionWithRandomSeed {
+
+  private var seed: Expression = Literal(Uniform.random.nextLong(), LongType)
```

Comment:
hmm, why do you need `var`. Could you follow the existing approach, and define a case class w/ 3 parameters + constructors with 2 params.

Reply:
Please see above reply.